### PR TITLE
fix: classify batched DDL statements

### PR DESCRIPTION
**Pull request by larkee**
Currently an SQL statement such as `"CREATE TABLE t (...); INSERT INTO T (...)"` would be classified as batched DDL despite not all the statements being DDL.

This PR adds a classification check for each of the statements and raises an error if any of the batched statements are non-DDL.

### CODE REVIEW

File: google/cloud/spanner_dbapi/cursor.py [hunk 1]
```diff
@@ -174,10 +174,14 @@ def execute(self, sql, args=None):
...
             if classification == parse_utils.STMT_DDL:
+                ddl_statements = []
                 for ddl in sql.split(";"):
                     ddl = ddl.strip()
                     if ddl:
-                        self.connection._ddl_statements.append(ddl)
+                        if parse_utils.classify_stmt(ddl) != parse_utils.STMT_DDL:
+                            raise ValueError("Only DDL statements may be batched.")
+                        ddl_statements.append(ddl)
+                self.connection._ddl_statements.extend(ddl_statements)
```
Comment (IlyaFaer) [inline google/cloud/spanner_dbapi/cursor.py:184]: Maybe I'm missing something, but I don't see strong reasons to initiate `ddl_statements` and then extend `self.connection._ddl_statements`. I assume ddls can be appended into `self.connection._ddl_statements` directly.

**Review by IlyaFaer — APPROVED** [review]
LGTM

File: google/cloud/spanner_dbapi/cursor.py (hunk 1 above)
Comment (larkee) [reply google/cloud/spanner_dbapi/cursor.py:184]: If one statement is invalid, none of the statements is executed. In the case that there is a non-DDL statement, I think it's less confusing to queue none of the statements than to queue the statements up to the non-DDL one. WDYT?

Comment (IlyaFaer) [reply google/cloud/spanner_dbapi/cursor.py:184]: Ah, if to add first `n` ddls and fail than the next `execute()` call will execute these first `n` ddls as well. Which would be quiet unexpected. Yeah, that's a good catch!

**Pull request merged**